### PR TITLE
EVG-5313 reset all tasks in single host task groups

### PR DESCRIPTION
**Pull request by ablack12**
As compared to the original iteration of this: https://github.com/evergreen-ci/evergreen/commit/7f1c714ea371d492f9bcd18e7d241b443549070a

Update: this now queries directly from tasks, avoiding the project step altogether. If BuildId is granular enough then we can go forward, otherwise we'll block on https://jira.mongodb.org/browse/MCI-3200.

### CODE REVIEW

File: model/task_lifecycle.go [hunk 1]
```diff
@@ -228,9 +236,19 @@ func TryResetTask(taskId, user, origin string, detail *apimodels.TaskEndDetail)
...
+	if t.IsPartOfSingleHostTaskGroup() {
+		var tasks []task.Task
+		tasks, err = GetTasksInTaskGroup(t)
+		if err != nil {
+			return errors.Wrapf(err, "couldn't get tasks in group '%s'", t.TaskGroup)
+		}
+		if err = resetManyTasks(tasks, user); err != nil {
+			return errors.WithStack(err)
+		}
+	} else if err = resetTask(t.Id, user); err != nil {
```
Comment (tychoish) [inline model/task_lifecycle.go:248]: why are resetManyTasks and restTasks not methods on a task?

Comment (ablack12) [reply model/task_lifecycle.go:248]: resetTask has been here for a while, but I believe the things it includes couldn't be in the task package (it involves updating builds and versions as well).

Comment (tychoish) [reply model/task_lifecycle.go:248]: alas.

**Comment by ablack12** [comment]
Description updated

**Comment by ablack12** [comment]
Closing for now.

**Pull request closed without merging**